metals/src/main/scala/scala/meta/internal/metals/MtagsResolver.scala

```diff
@@ -0,0 +1,64 @@
+package scala.meta.internal.metals
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.util.control.NonFatal
+
+import coursierapi.error.SimpleResolutionError
+
+class MtagsResolver {
```

Comment:
Could this be an object? Since we are anyway using `ConcurrentHashMap`

That would make sure that we are always using the same resolver.

Reply:
I don't think that it's good idea :smile: 
Then this state will be attached to JVM instance instead of LanguageServer.

Reply:
Isn't de facto the case? `MtagsResolver` is only created once really in each JVM.

Reply:
I think it better to keep it per-server-instance rather that per-JVM-instance to avoid cache invalidation issue that is a classical programming challenge. 
I understand that it won't be an issue right now. However, I've seen the similar things too many times to get the strong feeling that it should be avoided.

For example. In case if we wouldn't have `fork := true` for tests, then it will be a problem if tests were run as `sbt unit/test cross/test` in a single JVM.